File: 2108 find-first-palindromic-string-in-the-array/solution.cpp
class Solution {
public:
    string firstPalindrome(vector<string>& words) {
         for(string s:words) {
            if(isPal(s)) return s;
        }
        return "";
    }
    bool isPal(string s) {
        int l=0,r=s.size()-1;
        while(l<r) {
            if(s[l] != s[r]) return false;
            l++;
            r--;
        }
        return true;
    }
};

File: 2108 find-first-palindromic-string-in-the-array/solution.rs
impl Solution {
    pub fn first_palindrome(words: Vec<String>) -> String {
        for s in words {
            if Self::isPal(&s) {
                return s
            }
        }
        String::new() 
    }

    pub fn isPal(s: &String) -> bool {
        let bytes = s.as_bytes();
        let mut l = 0;
        let mut r = bytes.len().saturating_sub(1);
        while l < r {
            if bytes[l] != bytes[r] {
                return false;
            }
            l += 1;
            r -= 1;
        }
        true
    }
}
